versioned_docs/version-8.6/self-managed/setup/guides/secret-management.md

```diff
@@ -0,0 +1,356 @@
+> - `webModeler.restapi.mail.existingSecret`
+> - `connectors.inbound.auth.existingSecret`
+> - `global.license.existingSecret`
+
+```yaml
+global:
+  secrets:
+    autoGenerated: false
+    name: "camunda-platform-secrets"
+  identity:
+    auth:
+      admin:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-admin-client-password"
+      connectors:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-connectors-client-password"
+      console:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-console-client-password"
+      operate:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-operate-client-password"
+      tasklist:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-tasklist-client-password"
+      optimize:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-optimize-client-password"
+      zeebe:
+        existingSecret:
+          name: "camunda-platform-secrets"
+        existingSecretKey: "identity-zeebe-client-password"
+identity:
+  firstUser:
+    existingSecret: "camunda-platform-secrets"
+    existingSecretKey: "identity-user-password"
+identityPostgresql:
+  auth:
+    existingSecret: "camunda-platform-secrets"
+identityKeycloak:
+  auth:
+    existingSecret: "camunda-platform-secrets"
+    passwordSecretKey: "identity-keycloak-admin-password"
+  postgresql:
+    auth:
+      existingSecret: "camunda-platform-secrets"
+      secretKeys:
+        adminPasswordKey: "identity-keycloak-postgresql-admin-password"
+        userPasswordKey: "identity-keycloak-postgresql-user-password"
+# WebModeler Database.
+postgresql:
+  enabled: true
+  auth:
+    existingSecret: "camunda-platform-secrets"
+    secretKeys:
+      adminPasswordKey: "webmodeler-postgresql-admin-password"
+      userPasswordKey: "webmodeler-postgresql-user-password"
+```
+
+---
+
+## Disabling Auto-Generated Secrets
+
+> 🛠️ **Troubleshooting:** If a component fails to start after disabling auto-generated secrets, double-check:
+>
+> - The secret exists in the correct namespace
+> - The `existingSecretKey` is accurate and matches the Kubernetes secret
+> - `autoGenerated: false` is set and all required secrets are provided
+
+By default, the Camunda Helm Chart auto-generates secrets like OAuth client passwords on first install. This ensures the platform can start without manual configuration but can lead to issues on upgrades if the secrets are regenerated.
+
+To disable auto-generation and use your own pre-created secrets, set:
+
+```yaml
+global:
+  secrets:
+    autoGenerated: false
+```
+
+This tells the chart to stop generating secrets and instead expect you to provide all necessary secrets via `existingSecret` references.
```

Comment:
```suggestion
This will stop auto-generating secrets. Ensure:
- All required secrets are created
- The `existingSecretKey` values match the keys in the Kubernetes Secret
- Secrets exist in the correct namespace
```